parler_tts/modeling_parler_tts.py

```diff
@@ -2786,6 +2905,19 @@ def _prepare_decoder_input_ids_for_generation(
                 )
                 model_kwargs["decoder_attention_mask"] = decoder_attention_mask
 
+        if not self.prompt_cross_attention:
+            prompt_hidden_states = model_kwargs["prompt_hidden_states"]
+            num_codebooks = self.decoder.num_codebooks
+            input = decoder_input_ids.reshape(-1, num_codebooks, decoder_input_ids.shape[-1])
+            inputs_embeds = sum(
+                [
+                    self.decoder.model.decoder.embed_tokens[codebook](input[:, codebook])
+                    for codebook in range(num_codebooks)
+                ]
+            )
+            inputs_embeds = torch.cat([prompt_hidden_states, inputs_embeds], dim=1)
+            model_kwargs["inputs_embeds"] = inputs_embeds
+
```

Comment:
Not sure to follow ! You prepend here, which makes total sense, but still leaves `prompt_hidden_states` in model_kwargs, so isn't it supposed to be used again in the forward pass?

Reply:
We prepend here to make sure that we have `"inputs_embeds"` in the model_kwargs when `_get_initial_cache_position` is called. I have not removed `prompt_hidden_states` from the `model_kwargs` here since I wanted to change as little as possible things from the current logic. Indeed, in the current implementation of `prepare_inputs_for_generation`, `decoder_inputs_embeds` (that corresponds to the inputs_embeds here created with prepending) is not handled and I have not changed that. Were we to remove `promp_hidden_states` from the `model_kwargs` at this stage, we need to then handle `decoder_inputs_embeds` in `prepare_inputs_for_generation`, which is doable, yet I felt the impact negligible and preferred to aim for a version that would change as little as possible things from the current logic.